libs/dal/src/repositories/subscriber-preference/subscriber-preference.entity.ts

```diff
@@ -13,14 +13,21 @@ export class SubscriberPreferenceEntity {
 
   _subscriberId: string;
 
-  _templateId: string;
+  _templateId?: string;
 
   enabled: boolean;
 
   channels: IPreferenceChannels;
+
+  level: PreferenceLevelEnum;
 }
 
 export type SubscriberPreferenceDBModel = ChangePropsValueType<
   SubscriberPreferenceEntity,
   '_environmentId' | '_organizationId' | '_subscriberId' | '_templateId'
 >;
+
+export enum PreferenceLevelEnum {
+  GLOBAL = 'GLOBAL',
+  TEMPLATE = 'TEMPLATE',
```

Comment:
Tend to agree, because today we could say if level is undefined it is most definitely PreferenceLevelEnum.TEMPLATE and hack it, but if we add in the future another level we will never know what the undefined means.